src/Shared/ParameterBindingMethodCache.cs

```diff
@@ -249,6 +249,60 @@ public bool HasTryParseMethod(ParameterInfo parameter)
             return (method?.Invoke(parameter), paramCount);
         }
 
+        private static MethodInfo? GetStaticMethodFromHierarchy(Type type, string name, Type[] parameterTypes, Type? returnType = null)
+        {
+            bool IsMatch(MethodInfo? method) => method is not null && (returnType is null || method.ReturnType == returnType);
+
+            var methodInfo = type.GetMethod(name, BindingFlags.Public | BindingFlags.Static | BindingFlags.FlattenHierarchy, parameterTypes);
+
+            if (IsMatch(methodInfo))
+            {
+                return methodInfo;
+            }
+
+            var candidateInterfaceMethodInfo = default(MethodInfo);
+
+            // Check all interfaces for implementations. Fail if there are duplicates.
+            foreach (var implementedInterface in type.GetInterfaces())
+            {
+                var interfaceMethod = implementedInterface.GetMethod(name, BindingFlags.Public | BindingFlags.Static, parameterTypes);
```

Comment:
what happens in the case an interface implements another? Should we be passing `BindingFlags.FlattenHierarchy` here too or does it not apply to interfaces?

Reply:
Actually I don't think it matters as the call to `GetInterfaces` gets all interfaces inherited or implemented so that effectively flattens the interface hierarchy up front.

Reply:
I now see you even have a test for this, lovely!